modules/machineset-capacity-reservation.adoc

```diff
@@ -0,0 +1,149 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating_machinesets/creating-machineset-azure.adoc
+// * machine_management/control_plane_machine_management/cpmso_provider_configurations/cpmso-config-options-azure.adoc
+
+ifeval::["{context}" == "creating-machineset-azure"]
+:azure:
+endif::[]
+ifeval::["{context}" == "cpmso-config-options-azure"]
+:azure:
+endif::[]
+ifeval::["{context}" == "creating-machineset-aws"]
+:aws:
+endif::[]
+ifeval::["{context}" == "cpmso-config-options-aws"]
+:aws:
+endif::[]
+
+:_mod-docs-content-type: PROCEDURE
+[id="machineset-capacity-reservation_{context}"]
+ifdef::azure[= Configuring Capacity Reservation by using machine sets]
+ifdef::azure[= Configuring Capacity Blocks by using machine sets]
+
+{product-title} version {product-version} and later supports using
+ifdef::azure[on-demand Capacity Reservation with Capacity Reservation groups on {azure-full} clusters.]
+ifdef::aws[Capacity Blocks on {aws-full} clusters.]
+
+You can configure a machine set to deploy machines on any available resources that match the parameters of a capacity request that you define.
+These parameters specify the VM size, region, and number of instances that you want to reserve.
+If your 
+ifdef::azure[{azure-short} subscription quota]
+ifdef::aws[Capacity Block purchase]
+can accommodate the capacity request, the deployment succeeds.
+
+For more information, including limitations and suggested use cases for this 
+ifdef::azure[{azure-short} offering, see link:https://learn.microsoft.com/en-us/azure/virtual-machines/capacity-reservation-overview[On-demand Capacity Reservation] in the {azure-full} documentation.]
+ifdef::aws[{aws-short} offering, see link:https://docs.aws.amazon.com/en_us/AWSEC2/latest/UserGuide/capacity-blocks-using.html#capacity-blocks-purchase[Work with Capacity Blocks] in the {aws-short} documentation.]
+
+ifdef::azure[]
+[NOTE]
+====
+You cannot change an existing Capacity Reservation configuration for a machine set.
+To use a different Capacity Reservation group, you must replace the machine set and the machines that the previous machine set deployed.
+====
```

Comment:
I did not see anything similar to this in the AWS Jira artifacts, please let me know if I missed something.